### PR TITLE
Update byol_module.py

### PR DESCRIPTION
**Pull request by albernsrya**
Bug risk

Write a detailed description of your PR here.

Please follow the guidelines for PRs contained [here](docs/pull_requests.md). Checklist:

- [x] Ensure that your PR is small, and implements one change.
- [x] Add unit tests for all functions that you introduced or modified.
- [x] Run PyCharm's code cleanup tools on your Python files.
- [ ] Link the correct GitHub issue for tracking.
- [x] Update the [Changelog](CHANGELOG.md) file: Describe your change in terms of 
Added/Changed/Removed/... in the "Upcoming" section.
- [x] When merging your PR, replace the default merge message with a description of your PR,
and if needed a motivation why that change was required.

### CODE REVIEW

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py [hunk 1]
```diff
@@ -141,11 +141,13 @@ def configure_optimizers(self) -> Any:
     def exclude_from_wt_decay(self,
                               named_params: Iterator[Tuple[str, T]],
                               weight_decay: float,
-                              skip_list: List[str] = ['bias', 'bn']) -> List[Dict[str, Any]]:
+                              skip_list: List[str] = None) -> List[Dict[str, Any]]:
```
Comment (ant0nsc) [inline InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:144]: The type annotation is no longer correct, it should now be Optional[List[str]]

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py [hunk 2]
```diff
@@ -141,11 +141,13 @@ def configure_optimizers(self) -> Any:
...
         """
         Convolution-Linear bias-terms and batch-norm parameters are excluded from l2-norm weight decay regularisation.
         https://arxiv.org/pdf/2006.07733.pdf Section 3.3 Optimisation and Section F.5.
         """
+        if skip_list is None:150
```
Comment (ant0nsc) [inline InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:149]: What's the 150 doing here?

**Review by ant0nsc — CHANGES_REQUESTED** [review]
Thanks for that change! tiny suggested changes please.

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py (hunk 1 above)
Comment (ant0nsc) [reply InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:144]: Also, it would be great to add the default value for skip list to docu

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py (hunk 2 above)
Comment (albernsrya) [reply InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:149]: thank you for plying, Do not use a mutable like list or dictionary as a default value to an argument. Python’s default arguments are evaluated once when the function is defined. Using a mutable default argument and mutating it will mutate that object for all future calls to the function as well. thank you

Comment (ant0nsc) [reply InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:149]: Yes, I get the reason for your PR, that's all fine. Can you clarify what the `:150` means?

Comment (albernsrya) [reply InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:149]: for lines 149 and 150 the same as line 144, but I added if skip_list is None: on line 149 and on line 150 I simplified it so that it is easy to read and not mutate (bug) thank you

Comment (ant0nsc) [reply InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:149]:
```suggestion
        if skip_list is None:            
```

**Comment by ant0nsc** [comment]
Thanks @albernsrya for the updates, this is looking better now. Can you please ensure that the checks on the PR pass? Flake8 complains. Also, can you add documentation for the argument, that explains what the default values are?

**Comment by albernsrya** [comment]
thank you i have fixed it maybe there is another correction

**Comment by ant0nsc** [comment]
@albernsrya , before pushing changes, can you please run flake8 and mypy on your local box, to ensure that the basic checks pass? If you see that any of the actions show "red" status, fix those proactively, rather than waiting for me.

If you're passionate about that type of bug, consider installing this package in the repo: https://github.com/ebeweber/flake8-mutable - this would automate the process of finding issues. You would then ensure that the whole repository is free of mutable defaults, which would be a great contribution to the overall code quality 😃 If you decide to go down that route, please ensure everything passing in your fork before updating the PR.

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py [hunk 3]
```diff
@@ -68,8 +68,7 @@ def forward(self, x: T) -> T:  # type: ignore
     def cosine_loss(a: T, b: T) -> T:
         a = F.normalize(a, dim=-1)
         b = F.normalize(b, dim=-1)
-        neg_cos_sim = -(a * b).sum(dim=-1).mean()
-        return neg_cos_sim
+        return -(a * b).sum(dim=-1).mean()
```
Comment (ant0nsc) [inline InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:71]: This change is outside of the scope of this PR, please undo.

File: InnerEye/ML/SSL/lightning_modules/byol/byol_module.py [hunk 4]
```diff
@@ -91,11 +90,9 @@ def shared_step(self, batch: BatchType, batch_idx: int) -> T:
         with torch.no_grad():
             z_img1 = self.target_network.forward_until_predictor(img_1)
             z_img2 = self.target_network.forward_until_predictor(img_2)
-        loss = 0.5 * (self.cosine_loss(h_img1, z_img2.detach())
+        return 0.5 * (self.cosine_loss(h_img1, z_img2.detach())
```
Comment (ant0nsc) [inline InnerEye/ML/SSL/lightning_modules/byol/byol_module.py:93]: This change is outside of the scope of this PR, please undo.

**Comment by ant0nsc** [comment]
I'm closing this PR for now - feel free to re-activate if you get time to fix the remaining issues.
